### PR TITLE
fix: init widget form when close sidebar & fix refresh bug in modal

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [x] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
Please see order by commits

### CODE REVIEW

File: apps/web/src/services/dashboards/shared/dashboard-widget-input-form/DashboardWidgetInputForm.vue [hunk 1]
```diff
@@ -438,7 +438,10 @@ export default defineComponent<Props>({
         };
 
         /* sync to widget form store */
-        watch(() => state.schemaFormData, (schemaFormData) => {
+        watch(() => state.schemaFormData, (schemaFormData, before) => {
+            if (isEmpty(state.widgetOptionsJsonSchema)) return;
+            if (state.widgetOptionsJsonSchema.properties?.cost_group_by && !schemaFormData?.cost_group_by) return;
+            if (state.widgetOptionsJsonSchema.properties?.asset_group_by && !before?.asset_group_by) return;
```
Comment (yuda110) [inline apps/web/src/services/dashboards/shared/dashboard-widget-input-form/DashboardWidgetInputForm.vue:444]: description: I added it to prevent initializing the form when there is no `group by` at the very beginning.

File: apps/web/src/services/dashboards/widgets/_components/WidgetViewModeSidebar.vue [hunk 2]
```diff
@@ -107,7 +104,7 @@ const state = reactive({
 /* Util */
 const updateDashboardWidgetStore = () => {
     // update widget info in dashboard detail store
-    const widgetInfo: DashboardLayoutWidgetInfo = cloneDeep(widgetFormState.widgetInfo);
+    const widgetInfo = cloneDeep(widgetFormState.widgetInfo) as DashboardLayoutWidgetInfo;
```
Comment (sulmoJ) [inline apps/web/src/services/dashboards/widgets/_components/WidgetViewModeSidebar.vue:109]: Question: Why did you change type location??

**Review by sulmoJ — APPROVED** [review]
LGTM!!
I left a little question

File: apps/web/src/services/dashboards/widgets/_components/WidgetViewModeSidebar.vue (hunk 2 above)
Comment (yuda110) [reply apps/web/src/services/dashboards/widgets/_components/WidgetViewModeSidebar.vue:109]: There was a type error because the type of `widgetFormState.widgetInfo` is an **optional** `DashboardLayoutWidgetInfo`

**Review by skdud4659 — APPROVED** [review]
GOOD!
